Koji editor updates (#124)
madewithkoji/docs: src/asciidoc/docs/develop/testingTemplates.adoc

The preview pane enables you to preview several standard contexts inside the editor.

You can select from the following types of previews.

* *Default* – Displays what a template looks like when running.
The *Default* context is always enabled.

* *Remix* – Displays what a template looks like in edit mode.
A creator enters edit mode when they remix a template.
The *Remix* context is always enabled, even if <<instant-remixing#,instant remixing>> is not.

* *More* - Displays a menu of additional previews.
The additional previews include the following.

- *Admin*, *About*, and *Sticker* – Display previews of standard contexts, if you've enabled the associated entitlements.
For more on contexts, see <<withkoji-vcc-package#_context>>.

- *Thumbnail* – Displays what a template looks like when shared on social media.
The *Thumbnail* context is always enabled.

- *Receipt (seller)* and *Receipt (buyer)* – Displays previews of the transaction receipts for the buyer and seller.
These contexts are available if you've implemented <<withkoji-koji-iap-package#_iap,in-app purchases>> and enabled the dynamic receipts entitlement.

> * *Remote previews* - The preview pane provides options for previewing a template on different devices or in different environments.

Reviewer: Similar note as on the editor page. FYI, in the editor, all of the previews are running on a staging server (that's the dev environment for the editor).

- On a mobile device – Open a preview on any device by scanning a QR code.

- In a new tab – Open a preview on a staging server.

=== Writing to the console

You can test JavaScript code by using the `console.log` method, which prints to the console.

Although the `console.log` method can be implemented in the Koji editor, the results must be viewed in the console, which is only available when the template is run remotely, such as in a new browser tab.

> You can use the <<_using_remote_preview,remote preview>> option to run the template in a new tab.

Reviewer: This link is broken. Also, consider adding the instructions here instead of pointing to them. It will be a short sentence and save the reader some jumping around.

The preview pane enables you to preview several standard contexts inside the editor.

You can select from the following types of previews.

* *Default* – Displays what a template looks like when running.
The *Default* context is always enabled.

* *Remix* – Displays what a template looks like in edit mode.
A creator enters edit mode when they remix a template.
The *Remix* context is always enabled, even if <<instant-remixing#,instant remixing>> is not.

* *More* - Displays a menu of additional previews.
The additional previews include the following.

- *Admin*, *About*, and *Sticker* – Display previews of standard contexts, if you've enabled the associated entitlements.
For more on contexts, see <<withkoji-vcc-package#_context>>.

- *Thumbnail* – Displays what a template looks like when shared on social media.
The *Thumbnail* context is always enabled.

- *Receipt (seller)* and *Receipt (buyer)* – Displays previews of the transaction receipts for the buyer and seller.
These contexts are available if you've implemented <<withkoji-koji-iap-package#_iap,in-app purchases>> and enabled the dynamic receipts entitlement.

> * *Remote previews* - The preview pane provides options for previewing a template on different devices or in different environments.

Reviewer: ```suggestion
* *Remote previews* - The preview pane provides options for previewing a template outside of the Koji editor.
```
To match the "editing" page...

- On a mobile device – Open a preview on any device by scanning a QR code.
Click the QR code button image:QRcode.png[QR code button] in the preview pane to display the QR code.

- In a new tab – Open a preview on a staging server.
Click the remote staging button image:remote-staging.png[Remote staging button] in the preview pane to open the preview in a new tab.

=== Writing to the console

You can test JavaScript code by using the `console.log` method, which prints to the console.

> Although the `console.log` method can be implemented in the Koji editor, the results must be viewed in the console, which is only available when the template is run remotely, such as in a new browser tab.

Reviewer: ```suggestion
Although the `console.log` method can be implemented in the Koji editor, the results must be viewed in the console, which is only available when the template is run remotely.
```
Because it's a bit redundant with the next sentence now.

It allows you to follow the communication happening between the Koji platform and the template.

[NOTE]

> The debugger is only useful for debugging published templates.

Reviewer: This note is a bit confusing now. Let's clarify or remove it.

If a template is unpublished, no data is available to the debugger.

> From the Koji editor, you can open the debugger by clicking *Launch Debugger* in the preview pane.

Reviewer: Emphasize how to open the debugger from the project details, rather than from the editor. e.g.,

"To open the debugger, go to *Developer Portal > My projects* and click the project. Then, click *Debugger* on the overview page. You can also open the debugger from the Koji editor by clicking …"
